chemprop/cli/train.py

```diff
@@ -493,6 +493,24 @@ def process_train_args(args: Namespace) -> Namespace:
             "Please use `--checkpoint` with `--freeze-encoder` instead."
         )
 
+    if args.freeze_encoder and args.checkpoint is None:
+        raise ArgumentError(
+            argument=None,
+            message="`--freeze-encoder` can only be used when `--checkpoint` is used.",
+        )
+
+    if args.frzn_ffn_layers > 0:
+        if args.checkpoint is None and args.model_frzn is None:
+            raise ArgumentError(
+                argument=None,
+                message="`--frzn-ffn-layers` can only be used when `--checkpoint` or `--model-frzn` (depreciated in v2.1) is used.",
+            )
+        if args.checkpoint is not None and not args.freeze_encoder:
```

Comment:
```suggestion
        if not args.freeze_encoder:
```
Do you need to check if the checkpoint is not None?

Reply:
The `--model-frzn` flag is still keeped in v2.1. When using this flag, it is not necessary to use `--freeze_encoder` to freeze the MPNN. Once the `--model-frzn` flag is removed, we can then use the code you suggested here.